src/robot/libraries/BuiltIn.py

```diff
@@ -2945,6 +3004,43 @@ def _sleep_in_parts(self, seconds):
             if remaining <= 0:
                 break
             time.sleep(min(remaining, 0.01))
+    
+    async def async_sleep(self, time_, reason=None):
```

Comment:
As I commented above, I believe we shouldn't add async keywords to BuiltIn. A separate library that's first developed externally and possibly later made a standard library would be better.

In this particular case, I'd also like to know what's the use case for this keyword? In what context would you need to have an async sleep and not a normal sleep?